src/inabox/inabox-viewport.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {viewerForDoc} from '../viewer';
+import {Viewport, ViewportBindingDef} from '../service/viewport-impl';
+import {getServiceForDoc} from '../service';
+import {resourcesForDoc} from '../../src/resources';
+import {layoutRectLtwh} from '../layout-rect';
+import {Observable} from '../observable';
+import {MessageType} from '../../src/3p-frame';
+import {IframeMessagingClient} from '../../3p/iframe-messaging-client';
+import {dev} from '../log';
+
+/** @const {string} */
+const TAG = 'inabox-viewport';
+
+/**
+ * Implementation of ViewportBindingDef that works inside an non-scrollable
+ * iframe by listening to host doc for position and resize updates.
+ *
+ * @implements {ViewportBindingDef}
+ */
+export class ViewportBindingInabox {
+
+  /**
+   * @param {!Window} win
+   */
+  constructor(win) {
+    /** @const {!Window} */
+    this.win = win;
+
+    /** @private @const {!Observable} */
+    this.scrollObservable_ = new Observable();
```

Comment:
I think @chenshay had a pr that lazily create observable(). Do we also want to do that for inabox host?